### PR TITLE
CDAP-14549 fix macros in plugins of plugins

### PR DESCRIPTION
**Pull request by albertshau**
Use a macro evaluator when a plugin instantiates another plugin.
Also correctly scope the plugin id when a plugin is registered
using the plugin selector APIs.

### CODE REVIEW

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-RUT1642

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/plugin/PluggableFilterTransform.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.function.Predicate;
+import javax.annotation.Nullable;
+
+/**
+ * A Transform that delegates its logic to a plugin. Used to test plugins created by other plugins.
+ */
+@Plugin(type = "transform")
+@Name(PluggableFilterTransform.NAME)
+public class PluggableFilterTransform extends Transform<StructuredRecord, StructuredRecord> {
+  public static final String NAME = "pluggableTransform";
```
Comment (CuriousVini) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/plugin/PluggableFilterTransform.java]: nit: `PluggableTransform `

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/plugin/ValueFilter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * Filters out records where a field has a specific value. Used to test plugins created by other plugins.
+ */
+@Plugin(type = PluggableFilterTransform.FILTER_PLUGIN_TYPE)
+@Name(ValueFilter.NAME)
+public class ValueFilter implements Predicate<StructuredRecord> {
+  public static final String NAME = "valueFilter";
```
Comment (CuriousVini) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/plugin/ValueFilter.java]: nit: `ValueFilter `

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/plugin/PluggableFilterTransform.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+import javax.annotation.Nullable;
+
+/**
+ * A Transform that delegates its logic to a plugin. Used to test plugins created by other plugins.
+ */
+@Plugin(type = "transform")
```
Comment (CuriousVini) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/plugin/PluggableFilterTransform.java]: `type = Transform.PLUGIN_TYPE`

**Review by CuriousVini — APPROVED** [review]
minor comments. code changes lgtm 👍

**Pull request merged**